engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/StubbedFlinkProcessCompiler.scala

```diff
@@ -42,12 +42,20 @@ abstract class StubbedFlinkProcessCompiler(process: CanonicalProcess,
         sourceType -> stubbedDefinition
       }
 
+    def sourceDefForFragment(frag: FragmentInputDefinition): ObjectWithMethodDef = {
+      new StubbedFragmentInputDefinitionSource(processConfig, userCodeClassLoader).createSourceDefinition(frag)
+    }
+
+    val stubbedSourceForFragment: Seq[(String, ObjectWithMethodDef)] = process.allStartNodes.map(_.head.data).collect {
```

Comment:
Is it possible to have this pattern matching inside `TestFlinkProcessCompiler.prepareSourceFactory` ?

Reply:
It is impossible, forgot about this